src/main/scala/com/cognite/spark/datasource/AssetsRelation.scala

```diff
@@ -96,10 +96,18 @@ class AssetsRelation(config: RelationConfig)(val sqlContext: SQLContext)
   private val batchSize = config.batchSize.getOrElse(Constants.DefaultBatchSize)
 
   override def update(rows: Seq[Row]): IO[Unit] = {
-    val assetsItems = rows.map(r => fromRow[UpdateAssetsItemBase](r))
-    val updateAssetsItem = assetsItems.map(a => UpdateAssetsItem(a))
-
-    post(config, uri"${baseAssetsURL(config.project)}/update", updateAssetsItem)
+    val updateAssetsItems = rows
+      .map(r => fromRow[UpdateAssetsItemBase](r))
+      .map(a => UpdateAssetsItem(a))
+    updateAssetsItems
+      .filter(updateAssetItem =>
+        updateAssetItem.id match {
+          case Some(id) => false
+          case None => true
+      })
+      .foreach(a => throw new IllegalArgumentException("Got asset with null id."))
```

Comment:
"Assets must have an id when using update" or similar so the user knows more about what is happening.